Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Update/UpdateStreetcodeHandler.cs

```diff
@@ -0,0 +1,181 @@
+            _repositoryWrapper.StreetcodeRepository.Update(streetcodeToUpdate);
+
+            await _repositoryWrapper.SaveChangesAsync();
+
+            return 1;
+        }
+
+        private async Task UpdateStreetcodeArtsAsync(IEnumerable<StreetcodeArtCreateUpdateDTO> arts)
+        {
+            await UpdateEntitiesAsync(arts, _repositoryWrapper.StreetcodeArtRepository);
+        }
+
+        private async Task UpdateTimelineItemsAsync(StreetcodeContent streetcode, IEnumerable<TimelineItemCreateUpdateDTO> timelineItems)
+        {
+            var (toUpdate, toCreate, toDelete) = CategorizeItems<TimelineItemCreateUpdateDTO>(timelineItems);
+
+            var timelineItemsUpdated = new List<TimelineItem>();
+            foreach(var timelineItem in toUpdate)
+            {
+                var timelineItemToUpdate = _mapper.Map<TimelineItem>(timelineItem);
+                timelineItemToUpdate.HistoricalContextTimelines = null;
+                timelineItemsUpdated.Add(timelineItemToUpdate);
+                await UpdateEntitiesAsync(timelineItem.HistoricalContexts, _repositoryWrapper.HistoricalContextTimelineRepository);
+            }
+
+            streetcode.TimelineItems.AddRange(timelineItemsUpdated);
+            streetcode.TimelineItems.AddRange(_mapper.Map<IEnumerable<TimelineItem>>(toCreate));
+            _repositoryWrapper.TimelineRepository.DeleteRange(_mapper.Map<IEnumerable<TimelineItem>>(toDelete));
+        }
+
+        private async Task UpdateStreetcodeToponymAsync(IEnumerable<StreetcodeToponymUpdateDTO> toponymsUpdateDTOs, StreetcodeContent streetcodeContent)
+        {
+            // comming soon
+		}
+
+        private async Task UpdateStatisticRecordsAsync(IEnumerable<StatisticRecordUpdateDTO> records)
+        {
+            await UpdateEntitiesAsync(records, _repositoryWrapper.StreetcodeCoordinateRepository);
+        }
+
+        private async Task UpdateCategoryContentsAsync(IEnumerable<StreetcodeCategoryContentUpdateDTO> categoryContents)
+        {
+            await UpdateEntitiesAsync(categoryContents, _repositoryWrapper.StreetcodeCategoryContentRepository);
+        }
+
+        private async Task UpdateRelatedFiguresRelationAsync(IEnumerable<RelatedFigureUpdateDTO> relatedFigures)
+        {
+            await UpdateEntitiesAsync(relatedFigures, _repositoryWrapper.RelatedFigureRepository);
+        }
+
+        private async Task UpdatePartnersRelationAsync(IEnumerable<PartnersUpdateDTO> partners)
+        {
+            await UpdateEntitiesAsync(partners, _repositoryWrapper.PartnerStreetcodeRepository);
+        }
+
+        private async Task UpdateFactsAsync(IEnumerable<FactUpdateDTO> facts)
+        {
+            await UpdateEntitiesAsync(facts, _repositoryWrapper.FactRepository);
+        }
+
+        private async Task UpdateStreetcodeTagsAsync(IEnumerable<StreetcodeTagUpdateDTO> tags)
+        {
+            await UpdateEntitiesAsync(tags, _repositoryWrapper.StreetcodeTagIndexRepository);
+        }
```

Comment:
I think that it is unnecessary to create a new method for it because it only invokes another one and doesn't have another logic

Reply:
done